### PR TITLE
Remove state from Profile

### PR DESCRIPTION
**Pull request by aneyzberg**
## Description
Creates Adapters for Profile and removes any state that had been managed internally

## How Has This Been Tested?
Jest snapshot tests

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.test.js [hunk 1]
```diff
@@ -21,7 +21,9 @@ import React from 'react';
 
 import GlobalNav from './../GlobalNav';
 import TopNav from './TopNav';
-import Profile from './Profile';
+import ProfileComponent, {
+  ProfileAdapter
```
Comment (nfiniteset) [inline packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.test.js]: Doesn't look like you're using ProfileAdapter.

File: packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.js [hunk 2]
```diff
@@ -25,7 +25,9 @@ import HIGNodeList from '../../../HIGNodeList';
 import ProjectAccountSwitcherComponent, {
   ProjectAccountSwitcher
 } from './ProjectAccountSwitcher';
-import ProfileComponent, { Profile } from './Profile';
+import ProfileComponent, {
+  ProfileAdapter
```
Comment (nfiniteset) [inline packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.js]: Doesn't look like you're using ProfileComponent

File: packages/react-hig/src/elements/components/GlobalNav/TopNav/Profile.test.js [hunk 3]
```diff
@@ -20,7 +20,9 @@ import React from 'react';
 
 import GlobalNav from './../GlobalNav';
 import TopNav from './TopNav';
-import Profile from './Profile';
+import ProfileComponent, {
+  ProfileAdapter
+} from '../../../../adapters/ProfileAdapter';
```
Comment (nfiniteset) [inline packages/react-hig/src/elements/components/GlobalNav/TopNav/Profile.test.js]: Doesn't look like you're using ProfileAdapter

File: packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.js (hunk 2 above)
Comment (aneyzberg) [inline packages/react-hig/src/elements/components/GlobalNav/TopNav/TopNav.js]: this is used in child validation towards the bottom of the file.

**Pull request merged**